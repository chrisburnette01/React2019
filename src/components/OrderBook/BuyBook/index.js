import React from 'react';
import PriceCell from 'ComponentsGeneric/PriceCell';
import {Column} from 'react-virtualized';
import MarketDataTable, {ObservedCellRenderer} from '../../../components-generic/MarketDataTable';

const priceCellRenderer = ObservedCellRenderer(
    ({data:[price]}) =>  (
        <React.Fragment>
            <PriceCell price={price} isBuy={true} />
        </React.Fragment>
    )
);

const amountCellRenderer = ObservedCellRenderer(
    ({data:[,amount]}) =>  (
        <React.Fragment>
            {amount}
        </React.Fragment>
    )
);

const exchangeCellRenderer = ObservedCellRenderer(
    ({data:[,,exchange]}) =>  (
        <React.Fragment>
            {exchange}
        </React.Fragment>
    )
);

export default ({width, height, bids, rowHeight=25, rowCount}) => {
    const cellWidth = parseInt(width/3);
    return (
        <MarketDataTable
            disableHeader={true}
            height={height}
            width={width}
            rowHeight={rowHeight}
            marketDataMap={bids}
            rowCount={rowCount}
        >
            <Column 
                width={cellWidth}
                dataKey='price'
                label='Price'
                cellRenderer={priceCellRenderer}
            />
            
            <Column 
                width={cellWidth}
                dataKey='amount'
                label='Amount'
                cellRenderer={amountCellRenderer}
            />

            <Column 
                width={cellWidth}
                dataKey='exchange'
                label='Exchange'
                cellRenderer={exchangeCellRenderer}
            />            
        </MarketDataTable>
    )
}