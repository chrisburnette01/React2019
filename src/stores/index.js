import OrderBookStore from './OrderBook';
import RecentTradesStore from './RecentTrades';
import ThemeStore from './Theme';
import TradingType from './TradingType';
import SnackbarStore from './SnackbarStore';
import YourAccountStore from './YourAccountStore';
import ModalStore from './ModalStore';
import OrderHistory from './OrderHistory';
import InstrumentsStore from './InstrumentStore'
import SessionStore from './SessionStore';
import WindowSize from './WindowSize';
import GridStore from './GridStore';
import once from 'lodash.once';


const ORDERBOOK = 'OrderBook'; 
const RECENTTRADES = 'RECENTTRADES'; 
const THEME = 'Theme';
const TRADINGTYPE = 'TradingType';
const SNACKBARSTORE = 'SnackbarStore';
const YOURACCOUNTSTORE = 'YourAccountStore';
const MODALSTORE = 'ModalStore';
const ORDERHISTORY = 'OrderHistory';
const INSTRUMENTS = 'Instruments';
const WINDOWSIZE = 'WindowSize';
const GRIDSTORE = 'GridStore';


export const STORE_KEYS = {
    ORDERBOOK,
    YOURACCOUNTSTORE,
    RECENTTRADES,
    THEME,
    TRADINGTYPE,
    SNACKBARSTORE,
    ORDERHISTORY,
    INSTRUMENTS,
    MODALSTORE,
    WINDOWSIZE,
    GRIDSTORE,
}

export default once(() => {
    const instrumentStore = InstrumentsStore();
    const orderBookStore =  OrderBookStore(instrumentStore);
    const recentTradesStore = RecentTradesStore(instrumentStore);
    
    // not exposed.
    SessionStore(instrumentStore);

    return ({
        [STORE_KEYS.ORDERHISTORY]: OrderHistory(),
        [STORE_KEYS.YOURACCOUNTSTORE]: YourAccountStore(),
        [STORE_KEYS.INSTRUMENTS]: instrumentStore,
        [STORE_KEYS.ORDERBOOK]: orderBookStore,
        [STORE_KEYS.RECENTTRADES]: recentTradesStore,
        [STORE_KEYS.THEME]: ThemeStore(),
        [STORE_KEYS.TRADINGTYPE]: TradingType(),
        [STORE_KEYS.SNACKBARSTORE]: SnackbarStore(),
        [STORE_KEYS.MODALSTORE]: ModalStore(),
        [STORE_KEYS.WINDOWSIZE]: WindowSize(),
        [STORE_KEYS.GRIDSTORE]: GridStore(),
    }) 
});
