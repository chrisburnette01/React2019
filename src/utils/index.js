import math from '../lib/mathjs/math'

export const splitAmtOnDecimal = (amt) => {
    // whole number case; returns empty string as digitsAfterDecimal
    if(amt.toString() === parseFloat(amt).toFixed(0)){
        return [amt.toString(), '']
    }
    else{
        const [ digitsBeforeDecimal, digitsAfterDecimal ] = amt.toString().split('.');
        return [digitsBeforeDecimal, digitsAfterDecimal]
    }
};

export const roundToFixedNum = (amt, decimals) => {
    const roundedNum = math.round(amt, decimals);
    return roundedNum.toFixed(decimals)
};

export const fillUntil = (limit=0, mapFn) => {
    let i = -1;
    let items = [];
    while(++i < limit) items.push(mapFn(i));
    return items;
};

export const pageIsVisible = (() => {
    let hidden; 
    if (typeof document.hidden !== "undefined") { // Opera 12.10 and Firefox 18 and later support 
        hidden = "hidden";
    } 
    else if (typeof document.msHidden !== "undefined") {
        hidden = "msHidden";
    } else if (typeof document.webkitHidden !== "undefined") {
        hidden = "webkitHidden";
    }

    return () => !document[hidden];
})();