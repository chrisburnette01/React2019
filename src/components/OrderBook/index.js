import React from 'react';
import {AutoSizer} from 'react-virtualized';
import BuyBook from 'Components/OrderBook/BuyBook';
import SellBook from 'Components/OrderBook/SellBook';
import Spread from 'Components/OrderBook/Spread';
import {inject} from 'mobx-react';
import {STORE_KEYS} from '../../stores';
const {ORDERBOOK:STORE_KEY} = STORE_KEYS;

// material select is 36px; must be subtracted from height
const OrderBook = (
    {
        [STORE_KEY]:{
            bids,
            asks,
            spread,
        },
    }
) => {
    const rowHeight = 25;
    return (
        <AutoSizer>
            {
                ({height, width})=>{
                    const buyBookRowCount = parseInt(((height-40-40)/2)/rowHeight)
                    return (
                        <React.Fragment>
                           <SellBook
                                width={width}
                                height={(height/2)}
                                asks={asks}
                                rowHeight={rowHeight}
                                rowCount={parseInt((((height-60)/2))/rowHeight)}
                            />
                            <Spread 
                                height='40' 
                                spread={spread} 
                                width={width}
                            />
                            <BuyBook
                                width={width}
                                height={((height-40)/2)}
                                bids={bids}
                                rowHeight={25}
                                rowCount={buyBookRowCount}
                            />
                        </React.Fragment>
                    )
                }
            }
        </AutoSizer>
    );
}

export default inject(STORE_KEY)(OrderBook);
