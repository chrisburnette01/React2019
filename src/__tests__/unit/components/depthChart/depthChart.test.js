import {getSellDepthData, getBuyDepthData, partitionBuysSells, getInnerQuartileRangeOrders} from '../../../../stores/utils/storeUtils';
import {getDepthData} from '../../../../stores/OrderBook';

const Asks =  new Map([
    [0, ["0.055980", "38.535300", "COINHUB"]],
    [1, ["0.056020", "38.289900", "COINHUB"]],
    [2, ["0.056070", "31.610700", "COINHUB"]],
    [3, ["0.056130", "4.580000", "BITFINEX"]],
    [4, ["0.056160", "5.348332", "BITFINEX"]],
    [5, ["0.056180", "48.983591", "BITFINEX"]],
    [6, ["0.056190", "0.375525", "BITFINEX"]],
    [7, ["0.056210", "0.509501", "TIDEX"]],
    [8, ["0.056220", "0.450000", "BITFINEX"]],
    [9, ["0.056230", "33.108700", "BITFINEX"]],
    [10, ["0.056250", "15.000000", "BITFINEX"]],
    [11, ["0.056400", "0.725636", "BITTREX,CRYPTOPIA"]],
    [12, ["0.056410", "2.080000", "COINBENE"]],
    [13, ["0.056440", "0.527551", "CRYPTOPIA,OOOBTC"]],
    [14, ["0.056450", "3.301883", "COINBENE,OOOBTC"]],
    [15, ["0.056460", "1.075877", "LIVECOIN,OOOBTC"]],
    [16, ["0.056470", "5.355121", "CEXIO,LIQUI,LIVECOIN"]],
    [17, ["0.056480", "16.564850", "BITFINEX,CEXIO,LIVECOIN"]],
    [18, ["0.056490", "137.467738", "BITFINEX,ETHFINEX,HITBTC,KRAKEN,OOOBTC"]],
    [19, ["0.056500", "267.503401", "BINANCE,BITFINEX,CRYPTOPIA,ETHFINEX,GDAX,HITBTC,KRAKEN,LIQUI,YOBIT"]],
    [20, ["0.056510", "12.183912", "CRYPTOPIA,HUOBI"]],
    [21, ["0.056520", "131.170038", "BINANCE,BITFINEX,COINBENE,ETHFINEX,GDAX,GEMINI,KRAKEN,POLONIEX,QRYPTOS"]],
    [22, ["0.056530", "27.440091", "BIBOX,BINANCE,ETHFINEX,GDAX,HUOBI"]],
    [23, ["0.056540", "11.658855", "BIBOX,ETHFINEX,GDAX,HUOBI,KRAKEN,KUCOIN,POLONIEX"]],
    [24, ["0.056550", "198.709572", "BINANCE,BITFINEX,BITSTAMP,COINUT,ETHFINEX,GDAX,HUOBI,KRAKEN,KUCOIN,OOOBTC,POLONIEX"]],
    [25, ["0.056560", "48.414833", "BINANCE,BITTREX,GDAX,HUOBI,KRAKEN,KUCOIN,OKEX"]],
    [26, ["0.056570", "337.265529", "BITFINEX,ETHFINEX,GDAX,GEMINI,HUOBI,KRAKEN,KUCOIN"]],
    [27, ["0.056580", "117.590932", "BINANCE,BITFINEX,BITSTAMP,COINUT,ETHFINEX,GDAX,HUOBI,KRAKEN,LBANK,LIQUI,OKEX"]],
    [28, ["0.056590", "111.579327", "BINANCE,BITFINEX,BITSTAMP,BITTREX,ETHFINEX,GDAX,KRAKEN,LBANK,LIVECOIN,OKEX,POLONIEX"]],
    [29, ["0.056600", "144.119634", "BINANCE,BITFINEX,CEXIO,CRYPTOPIA,ETHFINEX,GDAX,HITBTC,HUOBI,KRAKEN,LBANK,OKEX,QRYPTOS"]],
    [30, ["0.056610", "166.945563", "BINANCE,BITSTAMP,EXMO,GDAX,KRAKEN,LIQUI,OOOBTC,POLONIEX"]],
    [31, ["0.056620", "60.917347", "BIBOX,BITFINEX,POLONIEX"]]
]);

const Bids = new Map([
    [0, ["0.058010", "3.200578", "CCEX"]],
    [1, ["0.058000", "0.519333", "CCEX"]],
    [2, ["0.057930", "0.010132", "CCEX"]],
    [3, ["0.057910", "0.141181", "CCEX"]],
    [4, ["0.057830", "0.011197", "CCEX"]],
    [5, ["0.057200", "0.004032", "CCEX"]],
    [6, ["0.057000", "2.316119", "CCEX"]],
    [7, ["0.056890", "0.006344", "CCEX"]],
    [8, ["0.056690", "0.230000", "ALLCOIN"]],
    [9, ["0.056680", "0.018000", "ALLCOIN"]],
    [10, ["0.056650", "0.012000", "ALLCOIN"]],
    [11, ["0.056630", "0.100000", "ALLCOIN"]],
    [12, ["0.056620", "0.067000", "ALLCOIN"]],
    [13, ["0.056590", "0.003600", "COINEGG"]],
    [14, ["0.056580", "30.000000", "BITSBLOCKCHAIN"]],
    [15, ["0.056560", "14.213778", "ALLCOIN,CCEX"]],
    [16, ["0.056510", "6.000000", "WEX"]],
    [17, ["0.056500", "6.505100", "BCEX,COINBENE,LUNO,THEROCKTRADING"]],
    [18, ["0.056490", "3.577169", "COINBENE,QRYPTOS"]],
    [19, ["0.056460", "2.645000", "OKEX,RIGHTBTC"]],
    [20, ["0.056450", "0.575161", "OKEX,RIGHTBTC"]],
    [21, ["0.056440", "5.544292", "GDAX,KRAKEN,OKEX,RIGHTBTC,WEX"]],
    [22, ["0.056430", "12.690000", "BITBANK,GDAX,KRAKEN"]],
    [23, ["0.056420", "0.106895", "WEX"]],
    [24, ["0.056410", "4.590461", "GDAX,HUOBI,KRAKEN,OKEX,RIGHTBTC"]],
    [25, ["0.056400", "52.604524", "BITBANK,BITFINEX,BITSTAMP,CCEX,CRYPTOPIA,ETHFINEX,HUOBI,KRAKEN,LUNO"]],
    [26, ["0.056390", "87.890628", "BITFINEX,BITTREX,ETHFINEX,EXMO,GDAX,HUOBI,KRAKEN,QRYPTOS,RIGHTBTC,WEX"]],
    [27, ["0.056380", "42.867022", "BITFINEX,COINBENE,ETHFINEX,GDAX,HUOBI,LBANK,LIQUI,OKEX,RIGHTBTC"]],
    [28, ["0.056370", "196.489507", "BINANCE,BITBANK,BITFINEX,BITZ,ETHFINEX,GDAX,HUOBI,KUCOIN,OKEX"]],
    [29, ["0.056360", "116.937606", "BINANCE,GDAX,HUOBI,KRAKEN,OOOBTC,YOBIT,ZBCOM"]],
    [30, ["0.056350", "243.824950", "BINANCE,BITFINEX,COINBENE,COINEX,ETHFINEX,GDAX,HUOâ€¦,KRAKEN,LIQUI,OKEX,OOOBTC,POLONIEX,RIGHTBTC,ZBCOM"]],
    [31, ["0.056340", "57.615363", "BIBOX,BINANCE,BITFINEX,BITTREX,ETHFINEX,GEMINI,HUOBI,KRAKEN,LIQUI,OKEX,POLONIEX,RIGHTBTC"]],
    [32, ["0.056330", "134.618913", "BITBANK,BITFINEX,BITLISH,BITTREX,ETHFINEX,GDAX,HITBTC,HUOBI,INDODAX,KRAKEN,LIQUI,OKEX,RIGHTBTC"]],
    [33, ["0.056320", "571.805039", "BINANCE,BITFINEX,BITLISH,BITSTAMP,COINUT,ETHFINEX,ODAX,KRAKEN,LBANK,LIQUI,OKEX,RIGHTBTC,YOBIT,ZBCOM"]],
    [34, ["0.056310", "162.867853", "BINANCE,ETHFINEX,GDAX,GEMINI,HUOBI,KRAKEN,LBANK,OKEX,OOOBTC,POLONIEX,THEROCKTRADING,WEX,YOBIT"]],
    [35, ["0.056300", "2.692719", "OOOBTC,QRYPTOS"]],
    [36, ["0.056250", "0.088887", "CRYPTOPIA"]]
]);

const AllOrders = [ 
    [ 0.05598, 38.5353, 'Sell' ],
    [ 0.05602, 38.2899, 'Sell' ],
    [ 0.05607, 31.6107, 'Sell' ],
    [ 0.05613, 4.58, 'Sell' ],
    [ 0.05616, 5.348332, 'Sell' ],
    [ 0.05618, 48.983591, 'Sell' ],
    [ 0.05619, 0.375525, 'Sell' ],
    [ 0.05621, 0.509501, 'Sell' ],
    [ 0.05622, 0.45, 'Sell' ],
    [ 0.05623, 33.1087, 'Sell' ],
    [ 0.05625, 15, 'Sell' ],
    [ 0.0564, 0.725636, 'Sell' ],
    [ 0.05641, 2.08, 'Sell' ],
    [ 0.05644, 0.527551, 'Sell' ],
    [ 0.05645, 3.301883, 'Sell' ],
    [ 0.05646, 1.075877, 'Sell' ],
    [ 0.05647, 5.355121, 'Sell' ],
    [ 0.05648, 16.56485, 'Sell' ],
    [ 0.05649, 137.467738, 'Sell' ],
    [ 0.0565, 267.503401, 'Sell' ],
    [ 0.05651, 12.183912, 'Sell' ],
    [ 0.05652, 131.170038, 'Sell' ],
    [ 0.05653, 27.440091, 'Sell' ],
    [ 0.05654, 11.658855, 'Sell' ],
    [ 0.05655, 198.709572, 'Sell' ],
    [ 0.05656, 48.414833, 'Sell' ],
    [ 0.05657, 337.265529, 'Sell' ],
    [ 0.05658, 117.590932, 'Sell' ],
    [ 0.05659, 111.579327, 'Sell' ],
    [ 0.0566, 144.119634, 'Sell' ],
    [ 0.05661, 166.945563, 'Sell' ],
    [ 0.05662, 60.917347, 'Sell' ],
    [ 0.05801, 3.200578, 'Buy' ],
    [ 0.058, 0.519333, 'Buy' ],
    [ 0.05793, 0.010132, 'Buy' ],
    [ 0.05791, 0.141181, 'Buy' ],
    [ 0.05783, 0.011197, 'Buy' ],
    [ 0.0572, 0.004032, 'Buy' ],
    [ 0.057, 2.316119, 'Buy' ],
    [ 0.05689, 0.006344, 'Buy' ],
    [ 0.05669, 0.23, 'Buy' ],
    [ 0.05668, 0.018, 'Buy' ],
    [ 0.05665, 0.012, 'Buy' ],
    [ 0.05663, 0.1, 'Buy' ],
    [ 0.05662, 0.067, 'Buy' ],
    [ 0.05659, 0.0036, 'Buy' ],
    [ 0.05658, 30, 'Buy' ],
    [ 0.05656, 14.213778, 'Buy' ],
    [ 0.05651, 6, 'Buy' ],
    [ 0.0565, 6.5051, 'Buy' ],
    [ 0.05649, 3.577169, 'Buy' ],
    [ 0.05646, 2.645, 'Buy' ],
    [ 0.05645, 0.575161, 'Buy' ],
    [ 0.05644, 5.544292, 'Buy' ],
    [ 0.05643, 12.69, 'Buy' ],
    [ 0.05642, 0.106895, 'Buy' ],
    [ 0.05641, 4.590461, 'Buy' ],
    [ 0.0564, 52.604524, 'Buy' ],
    [ 0.05639, 87.890628, 'Buy' ],
    [ 0.05638, 42.867022, 'Buy' ],
    [ 0.05637, 196.489507, 'Buy' ],
    [ 0.05636, 116.937606, 'Buy' ],
    [ 0.05635, 243.82495, 'Buy' ],
    [ 0.05634, 57.615363, 'Buy' ],
    [ 0.05633, 134.618913, 'Buy' ],
    [ 0.05632, 571.805039, 'Buy' ],
    [ 0.05631, 162.867853, 'Buy' ],
    [ 0.0563, 2.692719, 'Buy' ],
    [ 0.05625, 0.088887, 'Buy' ] 
];

describe('test', () => {
    //sort order is tested in each new function as the wrong sort in any function can break the chart
    
    test('IQR function returns the same Snapshot', () => {
        const iqr = getInnerQuartileRangeOrders(AllOrders)
        expect(iqr).toMatchSnapshot()
    });

    test('Data is partitioned back into orignal order buy and sells', () => {
        const iqr = getInnerQuartileRangeOrders(AllOrders)
        const [buys, sells] = partitionBuysSells(iqr, 'Buy', 'Sell')

        expect(
            buys.every((buyTriplet)=> buyTriplet[2] === 'Buy')
        ).toBe(true)

        buys.forEach((buyTriplet, idx) => {
            if(idx > 0){
                expect(buyTriplet[0]).toBeLessThanOrEqual(buys[idx-1][0])
            } 
        })

        expect(
            sells.every((sellTriplet)=> sellTriplet[2] === 'Sell')
        ).toBe(true)

        sells.forEach((sellTriplet, idx) => {
            if(idx > 0){
                expect(sellTriplet[0]).toBeGreaterThanOrEqual(sells[idx-1][0])
            } 
        })
    })

    test('Sell data comes in Low to High and cumulative size is done on that same order', () => {
        const iqr = getInnerQuartileRangeOrders(AllOrders)
        const [, sells] = partitionBuysSells(iqr, 'Buy', 'Sell');
        const sellDepthData = getSellDepthData(sells);

        sellDepthData.forEach(
            ({x}, i) => {
                if (i > 0) {
                    expect(x).toBeGreaterThanOrEqual(sellDepthData[i-1].x)
                }
            }
        )

        expect(sells[0][1]).toEqual(sellDepthData[0].y)

        let _size = 0;
        sells.forEach(
            ([, size], i) => {
                _size += size;
                expect(_size).toEqual(sellDepthData[i].y)
            }
        )        
    })

    test('Buy data comes in High to Low and cumulative size is done on that same order', () => {
        const iqr = getInnerQuartileRangeOrders(AllOrders)
        const [buys, ] = partitionBuysSells(iqr, 'Buy', 'Sell');
        const buyDepthData = getBuyDepthData(buys);
        buyDepthData.forEach(
            ({x}, i) => {
                if (i > 0) {
                    expect(x).toBeGreaterThanOrEqual(buyDepthData[i-1].x)
                }
            }
        )
        //buys first size is equal to buyDepthData last size
        expect(buys[0][1]).toEqual(buyDepthData[buyDepthData.length - 1].y)

        //computes size in reverse
        buyDepthData.reverse()
        let _size = 0;
        buys.forEach(
            ([, size], i) => {
                _size += size;
                expect(_size).toEqual(buyDepthData[i].y)
            }
        )        
    })

    test('Snapshot of market depth output is consistent based on inputs', () => {
        const result = getDepthData(Asks, Bids, 100)
        expect(result).toMatchSnapshot()
    })

    // buys and sells both low to high as highcharts requires it. Highcharts will reverse the buys.
    test('Market Depth returns buys ordered low to high; size is biggest to smallest', () => {
        const [sellDepthData] = getDepthData(Asks, Bids, 100)
        sellDepthData.forEach(
            ({x}, i) => {
                if (i > 0) {
                    expect(x).toBeGreaterThanOrEqual(sellDepthData[i-1].x)
                }
            }
        )
    })

    test('Market Depth returns sells ordered low to high; size is smallest to biggest', () => {
        const [, buyDepthData] = getDepthData(Asks, Bids, 100)
        buyDepthData.forEach(
            ({x}, i) => {
                if (i > 0) {
                    expect(x).toBeGreaterThanOrEqual(buyDepthData[i-1].x)
                }
            }
        )
    })

    test('Midmarket is avg of top buy and top sell', () => {
        const [sellDepthData, buyDepthData, midmarket] = getDepthData(Asks, Bids, 100)
        expect(midmarket).toBe(((buyDepthData[0].x + sellDepthData[0].x)/2).toFixed(8))
    })
});