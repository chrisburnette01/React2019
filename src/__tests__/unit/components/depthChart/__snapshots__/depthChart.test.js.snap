// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test IQR function returns the same Snapshot 1`] = `
Array [
  Array [
    0.05602,
    38.2899,
    "Sell",
  ],
  Array [
    0.05607,
    31.6107,
    "Sell",
  ],
  Array [
    0.05613,
    4.58,
    "Sell",
  ],
  Array [
    0.05616,
    5.348332,
    "Sell",
  ],
  Array [
    0.05618,
    48.983591,
    "Sell",
  ],
  Array [
    0.05619,
    0.375525,
    "Sell",
  ],
  Array [
    0.05621,
    0.509501,
    "Sell",
  ],
  Array [
    0.05622,
    0.45,
    "Sell",
  ],
  Array [
    0.05623,
    33.1087,
    "Sell",
  ],
  Array [
    0.05625,
    15,
    "Sell",
  ],
  Array [
    0.0564,
    0.725636,
    "Sell",
  ],
  Array [
    0.05641,
    2.08,
    "Sell",
  ],
  Array [
    0.05644,
    0.527551,
    "Sell",
  ],
  Array [
    0.05645,
    3.301883,
    "Sell",
  ],
  Array [
    0.05646,
    1.075877,
    "Sell",
  ],
  Array [
    0.05647,
    5.355121,
    "Sell",
  ],
  Array [
    0.05648,
    16.56485,
    "Sell",
  ],
  Array [
    0.05649,
    137.467738,
    "Sell",
  ],
  Array [
    0.0565,
    267.503401,
    "Sell",
  ],
  Array [
    0.05651,
    12.183912,
    "Sell",
  ],
  Array [
    0.05652,
    131.170038,
    "Sell",
  ],
  Array [
    0.05653,
    27.440091,
    "Sell",
  ],
  Array [
    0.05654,
    11.658855,
    "Sell",
  ],
  Array [
    0.05655,
    198.709572,
    "Sell",
  ],
  Array [
    0.05656,
    48.414833,
    "Sell",
  ],
  Array [
    0.05657,
    337.265529,
    "Sell",
  ],
  Array [
    0.05658,
    117.590932,
    "Sell",
  ],
  Array [
    0.05659,
    111.579327,
    "Sell",
  ],
  Array [
    0.0566,
    144.119634,
    "Sell",
  ],
  Array [
    0.05661,
    166.945563,
    "Sell",
  ],
  Array [
    0.05662,
    60.917347,
    "Sell",
  ],
  Array [
    0.05689,
    0.006344,
    "Buy",
  ],
  Array [
    0.05669,
    0.23,
    "Buy",
  ],
  Array [
    0.05668,
    0.018,
    "Buy",
  ],
  Array [
    0.05665,
    0.012,
    "Buy",
  ],
  Array [
    0.05663,
    0.1,
    "Buy",
  ],
  Array [
    0.05662,
    0.067,
    "Buy",
  ],
  Array [
    0.05659,
    0.0036,
    "Buy",
  ],
  Array [
    0.05658,
    30,
    "Buy",
  ],
  Array [
    0.05656,
    14.213778,
    "Buy",
  ],
  Array [
    0.05651,
    6,
    "Buy",
  ],
  Array [
    0.0565,
    6.5051,
    "Buy",
  ],
  Array [
    0.05649,
    3.577169,
    "Buy",
  ],
  Array [
    0.05646,
    2.645,
    "Buy",
  ],
  Array [
    0.05645,
    0.575161,
    "Buy",
  ],
  Array [
    0.05644,
    5.544292,
    "Buy",
  ],
  Array [
    0.05643,
    12.69,
    "Buy",
  ],
  Array [
    0.05642,
    0.106895,
    "Buy",
  ],
  Array [
    0.05641,
    4.590461,
    "Buy",
  ],
  Array [
    0.0564,
    52.604524,
    "Buy",
  ],
  Array [
    0.05639,
    87.890628,
    "Buy",
  ],
  Array [
    0.05638,
    42.867022,
    "Buy",
  ],
  Array [
    0.05637,
    196.489507,
    "Buy",
  ],
  Array [
    0.05636,
    116.937606,
    "Buy",
  ],
  Array [
    0.05635,
    243.82495,
    "Buy",
  ],
  Array [
    0.05634,
    57.615363,
    "Buy",
  ],
  Array [
    0.05633,
    134.618913,
    "Buy",
  ],
  Array [
    0.05632,
    571.805039,
    "Buy",
  ],
  Array [
    0.05631,
    162.867853,
    "Buy",
  ],
  Array [
    0.0563,
    2.692719,
    "Buy",
  ],
  Array [
    0.05625,
    0.088887,
    "Buy",
  ],
]
`;

exports[`test Snapshot of market depth output is consistent based on inputs 1`] = `
Array [
  Array [
    Object {
      "type": "Sell",
      "x": 0.05602,
      "y": 38.2899,
    },
    Object {
      "type": "Sell",
      "x": 0.05607,
      "y": 69.9006,
    },
    Object {
      "type": "Sell",
      "x": 0.05613,
      "y": 74.4806,
    },
    Object {
      "type": "Sell",
      "x": 0.05616,
      "y": 79.828932,
    },
    Object {
      "type": "Sell",
      "x": 0.05618,
      "y": 128.812523,
    },
    Object {
      "type": "Sell",
      "x": 0.05619,
      "y": 129.188048,
    },
    Object {
      "type": "Sell",
      "x": 0.05621,
      "y": 129.697549,
    },
    Object {
      "type": "Sell",
      "x": 0.05622,
      "y": 130.147549,
    },
    Object {
      "type": "Sell",
      "x": 0.05623,
      "y": 163.256249,
    },
    Object {
      "type": "Sell",
      "x": 0.05625,
      "y": 178.256249,
    },
    Object {
      "type": "Sell",
      "x": 0.0564,
      "y": 178.981885,
    },
    Object {
      "type": "Sell",
      "x": 0.05641,
      "y": 181.06188500000002,
    },
    Object {
      "type": "Sell",
      "x": 0.05644,
      "y": 181.589436,
    },
    Object {
      "type": "Sell",
      "x": 0.05645,
      "y": 184.891319,
    },
    Object {
      "type": "Sell",
      "x": 0.05646,
      "y": 185.967196,
    },
    Object {
      "type": "Sell",
      "x": 0.05647,
      "y": 191.322317,
    },
    Object {
      "type": "Sell",
      "x": 0.05648,
      "y": 207.887167,
    },
    Object {
      "type": "Sell",
      "x": 0.05649,
      "y": 345.35490500000003,
    },
    Object {
      "type": "Sell",
      "x": 0.0565,
      "y": 612.8583060000001,
    },
    Object {
      "type": "Sell",
      "x": 0.05651,
      "y": 625.042218,
    },
    Object {
      "type": "Sell",
      "x": 0.05652,
      "y": 756.212256,
    },
    Object {
      "type": "Sell",
      "x": 0.05653,
      "y": 783.6523470000001,
    },
    Object {
      "type": "Sell",
      "x": 0.05654,
      "y": 795.3112020000001,
    },
    Object {
      "type": "Sell",
      "x": 0.05655,
      "y": 994.0207740000001,
    },
    Object {
      "type": "Sell",
      "x": 0.05656,
      "y": 1042.4356070000001,
    },
    Object {
      "type": "Sell",
      "x": 0.05657,
      "y": 1379.7011360000001,
    },
    Object {
      "type": "Sell",
      "x": 0.05658,
      "y": 1497.2920680000002,
    },
    Object {
      "type": "Sell",
      "x": 0.05659,
      "y": 1608.8713950000001,
    },
    Object {
      "type": "Sell",
      "x": 0.0566,
      "y": 1752.991029,
    },
    Object {
      "type": "Sell",
      "x": 0.05661,
      "y": 1919.936592,
    },
    Object {
      "type": "Sell",
      "x": 0.05662,
      "y": 1980.853939,
    },
  ],
  Array [
    Object {
      "type": "Buy",
      "x": 0.05625,
      "y": 1757.187811,
    },
    Object {
      "type": "Buy",
      "x": 0.0563,
      "y": 1757.098924,
    },
    Object {
      "type": "Buy",
      "x": 0.05631,
      "y": 1754.406205,
    },
    Object {
      "type": "Buy",
      "x": 0.05632,
      "y": 1591.538352,
    },
    Object {
      "type": "Buy",
      "x": 0.05633,
      "y": 1019.7333130000001,
    },
    Object {
      "type": "Buy",
      "x": 0.05634,
      "y": 885.1144,
    },
    Object {
      "type": "Buy",
      "x": 0.05635,
      "y": 827.499037,
    },
    Object {
      "type": "Buy",
      "x": 0.05636,
      "y": 583.674087,
    },
    Object {
      "type": "Buy",
      "x": 0.05637,
      "y": 466.736481,
    },
    Object {
      "type": "Buy",
      "x": 0.05638,
      "y": 270.246974,
    },
    Object {
      "type": "Buy",
      "x": 0.05639,
      "y": 227.379952,
    },
    Object {
      "type": "Buy",
      "x": 0.0564,
      "y": 139.489324,
    },
    Object {
      "type": "Buy",
      "x": 0.05641,
      "y": 86.8848,
    },
    Object {
      "type": "Buy",
      "x": 0.05642,
      "y": 82.294339,
    },
    Object {
      "type": "Buy",
      "x": 0.05643,
      "y": 82.187444,
    },
    Object {
      "type": "Buy",
      "x": 0.05644,
      "y": 69.497444,
    },
    Object {
      "type": "Buy",
      "x": 0.05645,
      "y": 63.953152,
    },
    Object {
      "type": "Buy",
      "x": 0.05646,
      "y": 63.377991,
    },
    Object {
      "type": "Buy",
      "x": 0.05649,
      "y": 60.732991,
    },
    Object {
      "type": "Buy",
      "x": 0.0565,
      "y": 57.155822,
    },
    Object {
      "type": "Buy",
      "x": 0.05651,
      "y": 50.650722,
    },
    Object {
      "type": "Buy",
      "x": 0.05656,
      "y": 44.650722,
    },
    Object {
      "type": "Buy",
      "x": 0.05658,
      "y": 30.436944,
    },
    Object {
      "type": "Buy",
      "x": 0.05659,
      "y": 0.436944,
    },
    Object {
      "type": "Buy",
      "x": 0.05662,
      "y": 0.433344,
    },
    Object {
      "type": "Buy",
      "x": 0.05663,
      "y": 0.366344,
    },
    Object {
      "type": "Buy",
      "x": 0.05665,
      "y": 0.266344,
    },
    Object {
      "type": "Buy",
      "x": 0.05668,
      "y": 0.254344,
    },
    Object {
      "type": "Buy",
      "x": 0.05669,
      "y": 0.236344,
    },
    Object {
      "type": "Buy",
      "x": 0.05689,
      "y": 0.006344,
    },
  ],
  "0.05613500",
]
`;
